cclib/parser/qchemparser.py

```diff
@@ -1457,6 +1472,77 @@ def extract(self, inputfile, line):
                 grad = QChem.parse_matrix(inputfile, 3, self.natom, ncolsblock)
                 self.grads.append(grad.T)
 
+            if "Starting direction for rxn path" in line:
+                self.rpath_direction = int(line.split()[-1])
+                # This appears before any other IRC/reaction path output. If
+                # geometries appeared before this job, we assume they have
+                # nothing to do with the IRC and ignore them, except for the
+                # immediately preceding geometry, which is the starting step.
+                #
+                # We also handle the case that an IRC job has been performed
+                # previously, but there were some coordinates present between
+                # its final step and whatever we are about to do.
+                if not hasattr(self, "ircstatus"):
+                    self.ircstatus = [ccData.OPT_UNKNOWN for _ in range(len(self.atomcoords) - 1)]
+                else:
+                    nmissing = len(self.atomcoords) - len(self.ircstatus)
+                    if nmissing > 0:
+                        self.ircstatus.extend([ccData.OPT_UNKNOWN for _ in range(nmissing - 1)])
+                self.ircstatus.append(ccData.OPT_NEW + ccData.OPT_DONE)
+
+            if "Reaction path following" in line:
+                # When this section appears, the immediately preceding
+                # energy/geometry corresponds to the converged step along the
+                # reaction path, and the other steps are corrector/predictor
+                # steps.
+                tokens = next(inputfile).split()
+                assert len(tokens) == 10
+                assert self.rpath_direction is not None
+                coord_idx = len(self.atomcoords) - 1
+                ircvalues_step = {
```

Comment:
is there some guarantee of what fields will be populated in this dict for different IRC calculations? It seems like we are creating another metadata situation where the fields of this are not well defined. This might be necessary though

Reply:
Right now, I think the only thing that's important is some indication of which direction we're going (up or down the TS eigenmode). Other programs might only allow you to go in one direction, but I need a way of differentiating between the two since Q-Chem tries both.